### PR TITLE
:sparkles: [NPU] Add portable Gemma RMSNorm API

### PR DESCRIPTION
**Pull request by TallMessiWu**
## Motivation

Qwen3.5 uses Gemma RMSNorm, whose checkpoint weight is an offset with the logical scale `1 + weight`. `torch_npu.npu_gemma_rms_norm` is supported on Ascend 910 but has no Ascend 950 kernel.

This PR is the kernel half of the paired fix with sgl-project/sglang#32745 and keeps Qwen3.5 runnable on Ascend 950. SoC selection belongs to the target wheel build; SGLang calls stable APIs without detecting the SoC or choosing providers.

## Modifications

- provide stable APIs in every target wheel:
  - `sgl_kernel_npu.norm.gemma_rmsnorm.npu_gemma_rms_norm`
  - `sgl_kernel_npu.norm.add_rmsnorm_bias.add_gemma_rms_norm`
- stage target-specific providers at wheel build time (includes #734 by @wangyao-i):
  - providers live under `python/sgl_kernel_npu/target_providers/<target>/`; the relative module path is the registration key, so `target_providers/Ascend950/norm/gemma_rmsnorm.py` becomes `sgl_kernel_npu/norm/gemma_rmsnorm.py`
  - `build_tools/target_provider.py` stages exactly one target tree; unknown targets, missing trees and collisions with common modules fail the build
  - `SGL_KERNEL_NPU_BUILD_TARGET` is required when building the wheel; `build.sh` always exports it
  - `setup.py` keeps both `target_providers` and `build_tools` out of the packaged modules, and CI requires the 910 and 950 provider trees to expose the same modules
- perform provider selection entirely during wheel staging; the installed module has no runtime SoC query, provider branch, or fallback
- preserve the established `npu_gemma_rms_norm(...)->(output, rstd)` contract; use the native operator on 910 and ACLNN `npu_rms_norm(input, 1 + weight, eps)` on 950
- preserve the established `add_gemma_rms_norm(...)->(norm_output, residual_sum)` name, location, and return contract while replacing its slower Triton implementation with `npu_add_rms_norm(input, residual, 1 + weight, eps)`
- normalize the `910B`/`910`/`910C`/`950` build selectors, auto-detect the local SoC with `npu-smi` when none is given (hosts without an NPU fall back to `Ascend910_9382`), and export one canonical wheel target
- keep the canonical wheel target separate from the concrete CMake compiler target:
  - generic or detected Ascend 950 kernels builds compile with `Ascend950PR_9599`, the target of the 950 release package, so `csrc/CMakeLists.txt` selects `arch35` and the Ascend 950-only ops
  - explicit concrete `Ascend950PR_*` / `Ascend950DT_*` targets pass through unchanged
  - the full build rejects Ascend 950, because DeepEP and the kernels share one CMake SoC and DeepEP builds with the Ascend 910C value

## Paired architecture

```text
SGLang #32745
  -> ordinary: npu_gemma_rms_norm
       -> 910 wheel (Ascend 910B/910C): native Gemma ACLNN
       -> 950 wheel: standard RMSNorm ACLNN with 1 + weight
  -> residual: add_gemma_rms_norm
       -> all target wheels: add RMSNorm ACLNN with 1 + weight
```

The wheel is target-specific; using a 910 wheel on 950, or vice versa, is not supported.

## Validation

- build/provider unit tests (`test_build_targets.py`, `test_gemma_rmsnorm_provider.py`, `test_target_provider.py`): 26 passed
- build-script probes: `910` -> wheel `Ascend910`, CMake `Ascend910_9382`; `950` or a detected Ascend 950 -> wheel `Ascend950`, CMake `Ascend950PR_9599`; explicit `Ascend950PR_958b` passes through; the full build on Ascend 950 fails with a message to build DeepEP and the kernels separately
- Black, isort, clang-format 18.1.8, codespell, shell syntax, and `git diff --check` passed
- Ascend 950PR_958b candidate run with torch-npu `2.10.0.post2.dev20260704`: all 13 ordinary-path correctness cases passed
- Ascend 950PR_958b residual-add comparison: all 32 FP16/BF16 cases passed; residual sums matched exactly and normalized outputs passed `atol=2e-2, rtol=2e-2`
- still pending on Ascend 950 with the `Ascend950PR_9599` build: Qwen3.5 BF16 load/warmup/deterministic request
- Ascend 910B/910C native-provider smoke validation is still pending

## Performance / Accuracy

Measured on Ascend 950PR_958b with 50 warmups and 200 synchronized p50 iterations per case. Each 32-case matrix covers FP16/BF16, rows `1/16/128/512`, and hidden sizes `256/2048/4096/5120`.

Ordinary provider candidate:

- Triton geometric speedup versus `npu_rms_norm(input, 1 + weight)`: **-20.25%**
- worst Triton latency regression: **82.34%**
- result: **FAIL**; the 950 ordinary provider is fixed to ACLNN

Residual-add provider using the pre-existing production Triton kernel:

- geometric `Triton / ACLNN` p50 latency ratio: **1.7765x**
- ACLNN p50 latency reduction relative to Triton: **43.7%**
- correctness: **32/32 passed**, residual-sum max absolute error **0**
- normalized-output max absolute error: FP16 **0.0078125**, BF16 **0.125**; all cases passed `atol=2e-2, rtol=2e-2`
- result: **FAIL** for Triton; `add_gemma_rms_norm` now uses ACLNN on all supported targets

### CODE REVIEW

File: python/sgl_kernel_npu/sgl_kernel_npu/utils/npu_device.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+import torch_npu
+
+logger = logging.getLogger(__name__)
+
+
+class NpuDeviceFamily(Enum):
+    """Ascend device families relevant to kernel implementation selection."""
+
+    ASCEND_310P = "310p"
```
Comment (ping1jing2) [inline python/sgl_kernel_npu/sgl_kernel_npu/utils/npu_device.py]: maybe we don't need it here

File: python/sgl_kernel_npu/README.md [hunk 2]
```diff
@@ -6,7 +6,7 @@ SGLang Kernels NPU
 SGLang Kernels for Ascend NPU
 
 ## Software and hardware
-Supported Hardware Models: Atlas A3 Series Products
+Supported Hardware Models: Atlas A2/A3 Series Products and Ascend A5 for Gemma RMSNorm
```
Comment (ping1jing2) [inline python/sgl_kernel_npu/README.md]: use `Ascend 950 series` instead of `Ascend A5`

File: python/sgl_kernel_npu/sgl_kernel_npu/utils/npu_device.py [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+class NpuDeviceFamily(Enum):
+    """Ascend device families relevant to kernel implementation selection."""
+
+    ASCEND_910B = "ascend_910b"
+    ASCEND_910C = "ascend_910c"
+    ASCEND_950 = "ascend_950"
+    UNKNOWN = "unknown"
+
+
+def _family_from_soc_version(soc_version: int) -> NpuDeviceFamily:
```
Comment (VDV1985) [inline python/sgl_kernel_npu/sgl_kernel_npu/utils/npu_device.py]: Why we need separate logic for 910b & 910c?

File: python/sgl_kernel_npu/sgl_kernel_npu/utils/npu_device.py (hunk 1 above)
Comment (TallMessiWu) [reply python/sgl_kernel_npu/sgl_kernel_npu/utils/npu_device.py]: Removed.

File: python/sgl_kernel_npu/README.md (hunk 2 above)
Comment (TallMessiWu) [reply python/sgl_kernel_npu/README.md]: Done.

File: python/sgl_kernel_npu/sgl_kernel_npu/utils/npu_device.py (hunk 3 above)
Comment (TallMessiWu) [reply python/sgl_kernel_npu/sgl_kernel_npu/utils/npu_device.py]: Fixed.

**Review by ping1jing2 — DISMISSED** [review]
